gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/ApiRequestExecutorImpl.java

```diff
@@ -223,8 +223,23 @@ protected void recordErrorMetrics(Throwable error) {
      */
     @Override
     public void execute() {
-        // Strip apikey
-        stripApiKey();
+
+        //load the api data based on request
+        registry.getApi(request.getApiOrgId(), request.getApiId(), request.getApiVersion(),
+                (IAsyncResult<Api> apiResult) -> {
+                    if (apiResult.isSuccess()) {
+                        api = apiResult.getResult();
```

Comment:
This handler represents an async response being fulfilled. Execution will continue from L239 before the handler is even called. 

However, if I grok this correctly (entirely possible I've missed something): it tries to assign and use the API object it resolved on L231 on L238? However, there's no guarantee the object has been assigned by that point.

Reply:
I've verified this is incorrect. https://github.com/apiman/apiman/issues/2527